src/main/java/ch/njol/skript/effects/EffHealth.java

```diff
@@ -31,105 +31,110 @@
 import ch.njol.skript.lang.Effect;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.slot.Slot;
 import ch.njol.util.Kleenean;
 import ch.njol.util.Math2;
+import org.bukkit.entity.Damageable;
 import org.bukkit.entity.LivingEntity;
 import org.bukkit.event.Event;
 import org.bukkit.inventory.ItemStack;
 import org.eclipse.jdt.annotation.Nullable;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Damage/Heal/Repair")
 @Description("Damage/Heal/Repair an entity, or item.")
-@Examples({"damage player by 5 hearts",
-		"heal the player",
-		"repair tool of player"})
+@Examples({
+	"damage player by 5 hearts",
+	"heal the player",
+	"repair tool of player"
+})
 @Since("1.0")
 public class EffHealth extends Effect {
 
 	static {
 		Skript.registerEffect(EffHealth.class,
-			"damage %livingentities/itemtypes% by %number% [heart[s]] [with fake cause %-damagecause%]",
+			"damage %livingentities/itemtypes/slots% by %number% [heart[s]] [with fake cause %-damagecause%]",
 			"heal %livingentities% [by %-number% [heart[s]]]",
-			"repair %itemtypes% [by %-number%]");
+			"repair %itemtypes/slots% [by %-number%]");
 	}
 
-	@SuppressWarnings("NotNullFieldNotInitialized")
 	private Expression<?> damageables;
 	@Nullable
-	private Expression<Number> damage;
-	private boolean heal = false;
+	private Expression<Number> amount;
+	private boolean isHealing;
 
-	@SuppressWarnings({"unchecked", "null"})
+	@SuppressWarnings("unchecked")
 	@Override
-	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parser) {
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		if (matchedPattern == 0 && exprs[2] != null)
 			Skript.warning("The fake damage cause extension of this effect has no functionality, " +
 				"and will be removed in the future");
 
-		damageables = exprs[0];
+		this.damageables = exprs[0];
 		if (!LivingEntity.class.isAssignableFrom(damageables.getReturnType())) {
```

Comment:
Something's wonky here:
![image](https://github.com/SkriptLang/Skript/assets/10354869/60942923-ec7e-4582-8c1c-1cc8a62c8920)
I don't think this check handles expressionlists properly, This should have a returntype of Object, and the expressionlist as a whole can't be set to a single item, so it fails, when it should maybe be checking the changer for each element of the expressionlist? I'm not sure on the correct approach tbh.

Reply:
I'm not quite sure if this check is even needed in all fairness, I don't use the changer utils anymore I'll run a few test to see. While I believe this might have mostly existed for event-values I doubt it actually matters now.

Reply:
Alright so I've pushed a "fix" that removed the line in question, everything seems to work as intended.
```
command /testing:
  trigger:
    set {_1} to diamond sword with damage 500
    set player's tool to {_1}
    repair {_1}, (player's tool) by 250
    
on item damage:
    damage event-item by 500
```
Please feel free to test, once tested to work as you described I'll mark as resolved